### DISCUSSION TITLE
Gateway ip detect on log

### DISCUSSION THREAD

**Discussion by billouetaudrey** [dani-garcia/vaultwarden#2256] [Q&A]
### Subject of the issue

Hi, when I try anything bad password/login, it's detect my gateway ip on logs, I try it over LTE

### Deployment environment

* Install method: Docker image
Docker Armv7 version 1.23.1

* Clients used: 
 web vault

### Steps to reproduce
Just use LTE, chrome then enter wrong id

[2022-01-23 12:37:07.097][vaultwarden::api::identity][ERROR] Username or password is incorrect. Try again. IP: 192.168.0.254. 

I try to add var IP_HEADER = X-Real-IP in ENV settings ( portainer ) 

When vaultwarden started, it's say : [INFO] No .env file found.

Any idea :(

Thanks

**Comment by Masgalor** [comment]
I am not entirely sure if I do understand your problem completely, but I will try my best.

You have set up vaultwarden on a raspberry that is reachable from the internet and you are accessing it from your phone via mobile-data. Correct so far?

When you misstype your password while logging in from your phone, the incident gets logged, but it is not logging your mobile IP but the local IP of your gateway. Still correct?

Finally, you added `IP_HEADER = X-Real-IP` to the environment varibales for your vaultwarden-container, but this did not help.

My guess is that you did slightly misunderstand how those headers work.
You do not need to tell vaultwarden to act on those headers, you need to tell your gateway to add those headers in the first place.
To help you any further we need to know what components are between vaultwarden and the internet.

Did you have a look at the wikipage for proxy examples, it may be helpfull.
https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples

**Comment by billouetaudrey** [comment]
Hi, thanks for your help

Will try asap :)
